[CS2113-W13-4] TutorLink

RCPilot1604 opened the pull request:
NUS professors juggle admin tasks like tracking attendance and managing assignments, which detracts from their time for lesson preparation. Our app simplifies these processes by automating routine tasks and presenting concise, essential information, freeing up valuable time and enhancing their focus on teaching.

Mahesh1772 commented on src/test/java/tutorlink/command/DeleteGradeCommandTest.java:
```diff
@@ -0,0 +1,80 @@
+
+    private GradeList addGradeForTest(Parser parser, AppState appState, String componentName) {
+        //Create student
+        String line = "add_student i/A1234567X n/John Doe";
+        Command addStudentCommand = new AddStudentCommand();
+        String[] argumentPrefixes = addStudentCommand.getArgumentPrefixes();
+        HashMap<String, String> arguments = parser.getArguments(argumentPrefixes, line);
+        CommandResult result = addStudentCommand.execute(appState, arguments);
+        assertNotNull(result);
+        assertEquals("Student John Doe (A1234567X) added successfully!", result.toString());
+        assertEquals(appState.students.getStudentArrayList().size(), 1);
+
+        //Create component
+        String examName = componentName;
+        double examMaxScore = 100.0;
+        double examWeight = 50.0;
+        Exam exam = new Exam(examName,examMaxScore, examWeight);
+
+        appState.components.addComponent(exam);
+
+        //Add grade
+        HashMap<String, String> gradeArguments = new HashMap<>();
+
+        String matricNumber = "A1234567X";
+        String componentDescription = componentName;
+        String scoreNumber = "75.0";
+        gradeArguments.put("i/",matricNumber);
+        gradeArguments.put("c/", componentDescription);
+        gradeArguments.put("s/", scoreNumber);
+
+        Command addGradeCommand = new AddGradeCommand();
+        CommandResult gradeResult = addGradeCommand.execute(appState, gradeArguments);
+
+        //Test grade added
+        assertNotNull(gradeResult);
+        assertEquals(String.format(successMessage, scoreNumber, componentDescription, matricNumber),
+                gradeResult.toString());
+
+        return appState.grades;
+    }
```
`addGradeForTest()` is too long (>30 LOC) and doing multiple things, violating the "Avoid Long Methods" and the "Single Responsibility" principle. Consider breaking it down into smaller helper methods.

Mahesh1772 commented on src/test/java/tutorlink/command/DeleteGradeCommandTest.java:
```diff
@@ -0,0 +1,80 @@
+
+        //Create component
+        String examName = componentName;
+        double examMaxScore = 100.0;
+        double examWeight = 50.0;
```
Consider converting magic numbers and strings to constants, to increase maintainability.

Mahesh1772 commented on src/test/java/tutorlink/command/DeleteGradeCommandTest.java:
```diff
@@ -0,0 +1,80 @@
+
+    private GradeList addGradeForTest(Parser parser, AppState appState, String componentName) {
+        //Create student
+        String line = "add_student i/A1234567X n/John Doe";
```
Variable names could be more descriptive

Mahesh1772 commented on src/test/java/tutorlink/command/DeleteGradeCommandTest.java:
```diff
@@ -0,0 +1,80 @@
+        HashMap<String, String> arguments = parser.getArguments(argumentPrefixes, line);
+        CommandResult result = addStudentCommand.execute(appState, arguments);
+        assertNotNull(result);
+        assertEquals("Student John Doe (A1234567X) added successfully!", result.toString());
```
The success string could be added to the UI class

Mahesh1772 commented on src/test/java/tutorlink/command/FindStudentCommandTest.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    @Test
+    void execute_matric_one() {
```
Test method names don't follow the recommended format` featureUnderTest_testScenario_expectedBehavior()`. Consider renaming.

Mahesh1772 commented on src/test/java/tutorlink/command/FindStudentCommandTest.java:
```diff
@@ -0,0 +1,80 @@
+    private FindStudentCommand findCommand;
+    private AddStudentCommand addCommand;
+    private final AppState appState = new AppState();
+    private HashMap<String, String> arguments;
+    private CommandResult result;
```
Class-level variables result and arguments should be local to the methods that use them. This reduces the risk of state-related bugs in tests.

Mahesh1772 commented on src/test/java/tutorlink/command/FindStudentCommandTest.java:
```diff
@@ -0,0 +1,80 @@
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class FindStudentCommandTest {
```
Consider adding Javadocs

Mahesh1772 commented on src/main/java/tutorlink/lists/StudentList.java:
```diff
@@ -0,0 +1,84 @@
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class StudentList {
```
Consider adding class level Javadocs

Mahesh1772 commented on src/main/java/tutorlink/lists/StudentList.java:
```diff
@@ -0,0 +1,84 @@
+    private static final String ERROR_DUPLICATE_MATRIC_NUMBER_ON_ADD =
+            "Error! Student with Matric Number, %s, already exists in the list!";
+    private static final String TO_STRING_DELIMITER = "\n";
+    private ArrayList<Student> studentArrayList;
```
Consider using Set instead of List as this will make it impossible to add duplicate values

Mahesh1772 commented on src/main/java/tutorlink/lists/StudentList.java:
```diff
@@ -0,0 +1,84 @@
+
+    //Use Java String formatting to replace %s with matricNumber
+    private static final String ERROR_DUPLICATE_MATRIC_NUMBER_ON_ADD =
+            "Error! Student with Matric Number, %s, already exists in the list!";
```
Consider adding this as a constant to UI class and importing the class

Mahesh1772 commented on src/test/java/tutorlink/lists/ComponentListTest.java:
```diff
@@ -0,0 +1,84 @@
+    @BeforeEach
+    void setup() {
+        componentList = new ComponentList();
+        comp1 = new Assignment("Homework 1", 30, 0.1);
+        comp2 = new Assignment("Homework 2", 30, 0.1);
+        comp3 = new Exam("Finals", 100, 0.5);
```
Consider using more descriptive variable names

Mahesh1772 commented on src/test/java/tutorlink/lists/ComponentListTest.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    @Test
+    void toString_test() {
```
Ensure to  follow the format `featureUnderTest_testScenario_expectedBehavior()` like you have with others

Mahesh1772 commented on src/test/java/tutorlink/lists/ComponentListTest.java:
```diff
@@ -0,0 +1,84 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ComponentListTest {
```
Consider adding Javadocs

Mahesh1772 commented on src/main/java/tutorlink/lists/GradeList.java:
```diff
@@ -0,0 +1,104 @@
+        return filteredList;
+    }
+
+    public double calculateStudentGPA(String matricNumber, ComponentList componentList)
+            throws IncompleteGradesException {
+        ArrayList<Grade> studentGrades = gradeArrayList
+                .stream()
+                .filter(grade -> grade.getStudent().getMatricNumber().equals(matricNumber.toUpperCase()))
+                .collect(Collectors.toCollection(ArrayList::new));
+
+        ArrayList<Component> allComponents = componentList.findAllComponents();
+
+        if (studentGrades.isEmpty()) {
+            throw new StudentNotFoundException("No grades found for student: " + matricNumber);
+        }
+
+        boolean hasAllComponents = allComponents.stream()
+                .allMatch(component -> studentGrades.stream()
+                        .anyMatch(grade -> grade.getComponent().equals(component)));
+
+        if (!hasAllComponents) {
+            throw new IncompleteGradesException(
+                    "Cannot calculate GPA: Student " + matricNumber + " is missing grades for some components");
+        }
+
+        return studentGrades
+                .stream()
+                .mapToDouble(grade -> grade.getScore() * grade.getComponent().getWeight())
+                .sum();
+    }
```
Method is doing too many things (SLAP violation) - consider breaking into smaller methods. Magic number 0.07 in tax calculation could be a constant

Mahesh1772 commented on src/main/java/tutorlink/lists/GradeList.java:
```diff
@@ -0,0 +1,104 @@
+        this.gradeArrayList = new ArrayList<>();
+    }
+
+    public boolean deleteGrade(String matricNumber, String componentDescription) {
+        matricNumber = matricNumber.toUpperCase();
+        componentDescription = componentDescription.toLowerCase();
+        for (Grade grade : gradeArrayList) {
+            if (grade.getStudent().getMatricNumber().equals(matricNumber)
+                    && grade.getComponent().getName().equals(componentDescription)) {
+                return gradeArrayList.remove(grade);
+            }
+        }
+        return false;
+    }
```
Consider using early return pattern to reduce nesting

Mahesh1772 reviewed (commented):
Great job overall! The code is solid, and the functionality is well-implemented. I only had a few minor recommendations, mainly around Javadocs, using constants and naming, but nothing that detracts from the quality of the work. Thanks for the effort and attention to detail here!

yakultbottle commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
```
Should this header Design & implementation be removed, since there is a separate Implementation header below?

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
 
-{Describe the value proposition: what problem does it solve?}
+- At app launch, TutorLink initializes components (<code>Parser</code>, <code>Ui</code>, <code>Storage</code>, <code>
+  AppState</code>).
```
formatting issue here

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
 
-## Product scope
-### Target user profile
+![Architecture.png](diagrams/Architecture.png)
```
confusing diagram, do not understand what the arrows and lines are trying to say, maybe add a legend of what each line

yakultbottle commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+3. Add/Delete `Student` and `Component`.
+4. Return `CommandResult` that contains the result of the Add/Delete operation.
+
+The following sequence diagrams depict the exact steps involved in the `AddStudentCommand`:
+
+![AddStudentCommand.png](diagrams/AddStudentCommand.png)
+
```
![image](https://github.com/user-attachments/assets/e13d16e9-2d89-4167-9e5b-058788279680)

Should there be a dotted line indicating the main program calling this function?

yakultbottle commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+
+### Command execution sequence: 
+
+All commands follow the sequence as described in the diagram below: 
+
+![ArchitectureSequenceGrouped.png](diagrams%2FArchitectureSequenceGrouped.png)
+
```
![image](https://github.com/user-attachments/assets/50959538-3448-4225-b285-b2310f3dd854)

Could this constructor be omitted from this diagram, since it is not necessary to demonstrate command flow sequence?

wongwh2002 replied:
should you use alt for exception? maybe use alt [success] and [failure]?
would opt [success] be better

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+
+All commands follow the sequence as described in the diagram below: 
+
+![ArchitectureSequenceGrouped.png](diagrams%2FArchitectureSequenceGrouped.png)
```
should the displayException have a return to standardise?

wongwh2002 replied:
should you concise the alt [success] sequence? the (new StudentComponent, ComponentStorage, GradeStorage) and the rest is a bit messy/confusing

yakultbottle commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+
+The load list methods for the Storage classes have largely similar logic flows.  
+The following section and sequence diagram elaborate on the implementation of the `loadGradeList` method in `GradeStorage`:
+
+![GradeStorage.png](diagrams/GradeStorage.png)
+
```
Could all the instantiations be moved into a separate diagram?

yakultbottle reviewed (approved):
Overall looks good, only minor questions

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+
+The following sequence diagrams depict the exact steps involved in the `AddStudentCommand`:
+
+![AddStudentCommand.png](diagrams/AddStudentCommand.png)
```
should you have a lifeline for where "main"/ or something calls AddStudentCommand for better readability

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+
+Likewise, the sequence diagram for `DeleteStudentCommand` is as follows: 
+
+![DeleteStudentCommand.png](diagrams%2FDeleteStudentCommand.png)
```
should you have a lifeline for where "main"/ or something calls DeleteStudentCommand for better readability

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+
+The sequence diagram of the DeleteGradeCommand is shown below.
+
+![AddGradeCommand.png](diagrams%2FAddGradeCommand.png)
```
should AddStudentCommand have a X at the end of the lifeline of an object to show its deletion

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
+
+![GradeStorage.png](diagrams/GradeStorage.png)
+
+1. TutorLink constructs a new `GradeStorage`.
+2. TutorLink calls `loadGradeList`.
+3. `GradeStorage` creates a new `ArrayList` of `Grade`s.
+4. `GradeStorage` creates a new `Scanner` with the path to the file.
+5. While there are next lines in the data file:
+   - `Scanner` returns the current file line as a String and moves to the next file line.
+   - `GradeStorage` calls its `getGradeFromFileLine` method with the file line.
+   - If the file line references a valid `Component` and a valid `Student`, a `Grade` is returned and added to the `ArrayList`.
+   - If not (e.g. if data file was corrupted), the file line is simply ignored, and the loop continues to the next iteration.
+6. The `ArrayList` of `Grade`s is returned to TutorLink.
```
should this be more concise instead of going through your entire program/algorithm

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
```
should you include a class diagram

wongwh2002 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,189 @@
 
```
lgtm in general!

wongwh2002 reviewed (commented):
other than the comments, lgtm!

sarahchow03 commented on docs/diagrams/ArchitectureSequenceGrouped.png:
Should there be some standardization for the function calls in this diagram, specifically for `displayResult` to be `displayResult()`?

sarahchow03 commented on docs/diagrams/GradeStorage.png:
Should the lifeline for Scanner terminate after `X`, as it indicates the end of the lifeline?

sarahchow03 replied:
Minor nit: In the `alt` frame of this diagram, one or two words to indicate what was successful will be good!

sarahchow03 commented on docs/diagrams/DeleteStudentCommand.png:
Although including activation bars are not compulsory, with reference to the sequence diagram for `AddGradeCommand` and `DeleteGradeCommand`, should the activation bar for `:DeleteStudentCommand` be included to ensure uniformity across your team's diagrams?

sarahchow03 commented on docs/diagrams/FindStudentCommand.png:
Similar to a previous comment, the activation bar for :FindStudentCommand could be included to ensure uniformity across your team's diagrams!

sarahchow03 replied:
In the `alt` frame of this diagram, the second condition has not been labelled. This could be fixed by adding a `[else]`.

sarahchow03 commented on docs/diagrams/AddComponentCommand.png:
Similar to a previous comment, should the activation bar for `:AddComponentCommand`  be included to ensure uniformity across your team's diagrams?

sarahchow03 reviewed (commented):
Overall, the diagrams are well-documented and rigorous! I liked how the classes were colour-coded and standard across all diagrams, and that there was efficient use of reference frames in your team's DG. Good job!

yeekian committed:
Merge branch 'master' of https://github.com/yeekian/tp into branch-DeveloperGuide

yeekian committed:
Merge pull request #129 from yeekian/branch-DeveloperGuide

Complete Draft of Developer Guide, except for storage data in appendix E

ThienDuc3112 committed:
Merge branch 'AY2425S1-CS2113-W13-4:master' into calcGPA

ThienDuc3112 committed:
Fix ui test

ThienDuc3112 committed:
Fix ui test again

RCPilot1604 committed:
Modify .puml files

RCPilot1604 committed:
Modify ArchitectureSequenceGrouped.puml

RCPilot1604 committed:
Re-export png files

RCPilot1604 committed:
Modify DeveloperGuide.md

RCPilot1604 committed:
Merge pull request #130 from RCPilot1604/branch-DG-cleanup

Clean up DG

jinzihan2002 committed:
Merge pull request #116 from ThienDuc3112/calcGPA

Add calcGPA command

jinzihan2002 committed:
Update DG manual testing for Storage

jinzihan2002 committed:
Update GradeStorage uml

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-7-Nov

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Modify DeveloperGuide.md
  Re-export png files
  Modify ArchitectureSequenceGrouped.puml
  Modify .puml files
  Fix ui test again
  Fix ui test
  Fix codestyle
  Fix tests
  Handle totalWeighting = 0 case
  Fix updating GPA mechanic
  Add command

# Conflicts:
#	docs/diagrams/GradeStorage.puml

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

yeekian committed:
Merge branch 'master' of https://github.com/yeekian/tp into branch-refactor

yeekian committed:
Add print individual student grade list

yeekian committed:
Reformat output messages for grade list

RCPilot1604 committed:
Modify AddComponentCommand to include check for total weighting

RCPilot1604 committed:
Modify AddComponentCommand to check for total weighting

RCPilot1604 committed:
Modify Commons to add max weighting error message

RCPilot1604 committed:
Fix Component.equals()

RCPilot1604 committed:
Update IO redirection testing

RCPilot1604 committed:
Modify tests

RCPilot1604 committed:
Fix checkstyle and format

yeekian committed:
Add final GPA information for all students in grade list

yeekian committed:
Update list_grade in User Guide

jinzihan2002 committed:
Merge pull request #132 from yeekian/branch-refactor

Add more details to printed grade list to follow user guide

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

RCPilot1604 committed:
Fix IO redirection

yeekian committed:
Merge pull request #133 from RCPilot1604/branch-bugfix-component

Branch bugfix component

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

yeekian committed:
Merge branch 'master' of https://github.com/yeekian/tp into branch-Storage

jinzihan2002 committed:
Update setup and storage in DG to reflect new changes

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-7-Nov

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Fix IO redirection
  Update list_grade in User Guide
  Add final GPA information for all students in grade list
  Fix checkstyle and format
  Modify tests
  Update IO redirection testing
  Fix Component.equals()
  Modify Commons to add max weighting error message
  Modify AddComponentCommand to check for total weighting
  Modify AddComponentCommand to include check for total weighting
  Reformat output messages for grade list
  Add print individual student grade list

jinzihan2002 committed:
Update UG with QnA about editing data files

jinzihan2002 committed:
Merge pull request #134 from jinzihan2002/Zihan-7-Nov

Update UG and DG to reflect new changes to code

yeekian committed:
Refactor ListGradeCommand

yeekian committed:
Add load and save GPA field of student in StudentStorage

yeekian committed:
Add new constructor for Student object

yeekian committed:
Merge branch 'master' of https://github.com/yeekian/tp into branch-UserGuide

yeekian committed:
Refactor User Guide

jinzihan2002 committed:
Merge pull request #135 from yeekian/branch-Storage

Save GPA information in Storage

yeekian committed:
Merge pull request #137 from yeekian/branch-UserGuide

Refactor User Guide

RCPilot1604 committed:
Refactor GPA to Percentage Score

RCPilot1604 committed:
Add Multiple Query Result error message instead of duplicate

RCPilot1604 committed:
Refactor GPA to Percentage Score

RCPilot1604 committed:
Refactor GPA into Percentage Score

RCPilot1604 committed:
Refactor GPA to Percentage Score

RCPilot1604 committed:
Add Multiple Query Results Error Message

RCPilot1604 committed:
Convert GPA into percentage score

RCPilot1604 committed:
Refactor GPA into Percentage Score

RCPilot1604 committed:
Fix checkstyle and tests

RCPilot1604 committed:
Fix IO redirection

jinzihan2002 committed:
Add page breaks to format UG

RCPilot1604 committed:
Merge branch 'master' into branch-bugfix-GPA

jinzihan2002 committed:
Merge pull request #139 from jinzihan2002/Zihan-7-Nov-2

Add page breaks to format UG

RCPilot1604 committed:
Fix unmerged branch

RCPilot1604 committed:
Fix IO redirection

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

RCPilot1604 committed:
Merge pull request #138 from RCPilot1604/branch-bugfix-GPA

Branch bugfix GPA

ThienDuc3112 committed:
Fix #175

ThienDuc3112 committed:
Fix #176

ThienDuc3112 committed:
Fix #146

ThienDuc3112 committed:
Fix #154

ThienDuc3112 committed:
Checkstyle

RCPilot1604 committed:
Move totalWeight to AppState

RCPilot1604 committed:
Modify Add/DeleteComponentCommand to properly modify totalWeighting

RCPilot1604 committed:
Add MAX_WEIGHT to Commons

RCPilot1604 committed:
Remove totalWeight from Component

RCPilot1604 committed:
Update Add/DeleteComponent and ComponentList tests

RCPilot1604 committed:
Merge pull request #198 from ThienDuc3112/bugfix

Bug fix from PED (#175 , #176)

RCPilot1604 committed:
Merge branch 'master' into branch-bugfix-component-totalWeighting

# Conflicts:
#	src/main/java/tutorlink/command/AddComponentCommand.java

RCPilot1604 committed:
Modify totalWeighting

RCPilot1604 committed:
Add more tests to check addition of negative weighting

ThienDuc3112 committed:
Merge pull request #199 from RCPilot1604/branch-bugfix-component-totalWeighting

Branch bugfix component total weighting

ThienDuc3112 committed:
Update component command

ThienDuc3112 committed:
Minor code style change

ThienDuc3112 committed:
Fix command word

ThienDuc3112 committed:
Added unit test

ThienDuc3112 committed:
Fix method names

ThienDuc3112 committed:
More method name fixing

ThienDuc3112 committed:
More descriptive error

RCPilot1604 committed:
Merge pull request #200 from ThienDuc3112/UpdateComponentCommand

Update component command

jinzihan2002 committed:
Remove gpa from StudentStorage and calculate gpa after loading

jinzihan2002 committed:
Update Storage classes to strip whitespace from file lines

jinzihan2002 committed:
Update GradeStorage to reject duplicate grades when loading

TrungBui32 committed:
Update MD

TrungBui32 committed:
Merge pull request #202 from TrungBui32/md

Update MD

ThienDuc3112 committed:
Added update student gpa

ThienDuc3112 committed:
Added unit test

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-10-Nov

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Update MD

jinzihan2002 committed:
Update StudentStorage to check regex of matricNumber

jinzihan2002 committed:
Merge pull request #203 from ThienDuc3112/updateAllGpa

Added update student gpa

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-10-Nov

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Added unit test
  Added update student gpa

jinzihan2002 committed:
Update main to use updateAllStudentPercentageScores after loading

jinzihan2002 committed:
Update GradeStorage to reject scores < 0 or > maxScore

RCPilot1604 committed:
Add more descriptive messages to ListComponent and ListStudent command

jinzihan2002 committed:
Update ComponentStorage to reject invalid maxScores and weights

jinzihan2002 committed:
Merge pull request #201 from jinzihan2002/Zihan-10-Nov

Bug fix for Storage

RCPilot1604 committed:
Merge branch 'master' into branch-empty-list-message

RCPilot1604 committed:
Update total weighting error message

RCPilot1604 committed:
Update score > max_score error message

RCPilot1604 committed:
Further update score > max_score error message

jinzihan2002 committed:
Update text-ui-test to work with initial data

jinzihan2002 committed:
Fix runtest.bat

TrungBui32 committed:
Update ERROR_ILLEGAL_MATRIC_NUMBER

RCPilot1604 committed:
Add handling of multiple prefixes

jinzihan2002 committed:
Add help message

jinzihan2002 committed:
Update UG to match help message

RCPilot1604 committed:
Modify UG to reflect notes on command

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

RCPilot1604 committed:
Update UG

RCPilot1604 committed:
Add first wave of Javadocs

RCPilot1604 committed:
Add javadocs

RCPilot1604 committed:
Merge branch 'branch-JavaDoc' of https://github.com/RCPilot1604/tp into branch-JavaDoc

# Conflicts:
#	src/main/java/tutorlink/storage/ComponentStorage.java

RCPilot1604 committed:
Fix javadoc for ComponentList

RCPilot1604 committed:
Fix GradeStorage

yeekian committed:
Format AddComponentCommand to SLAP and add Javadocs

yeekian committed:
Format AddGradeCommand to SLAP and add Javadocs

yeekian committed:
Format DeleteGradeCommand to SLAP and add Javadocs

yeekian committed:
Merge pull request #210 from RCPilot1604/branch-JavaDoc

Add Javadocs

RCPilot1604 committed:
Merge pull request #211 from yeekian/branch-PEDryRunBugFix

Format Commands to SLAP and add Javadocs

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

RCPilot1604 committed:
Add validation to prevent excessively large max score from being added

yeekian committed:
Merge pull request #209 from jinzihan2002/ui-help-txt

Add help message and displayHelpMessage() UI method

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

yeekian committed:
Merge branch 'master' of https://github.com/yeekian/tp into branch-PEDryRunBugFix

RCPilot1604 committed:
Update AddComponentCommandTest to test new validation

RCPilot1604 committed:
Merge branch 'master' into branch-fix-193

# Conflicts:
#	src/main/java/tutorlink/parser/Parser.java

RCPilot1604 committed:
Merge pull request #207 from TrungBui32/matricError

Update ERROR_ILLEGAL_MATRIC_NUMBER

RCPilot1604 committed:
Merge branch 'master' into branch-fix-173

# Conflicts:
#	src/main/java/tutorlink/command/ListStudentCommand.java

yeekian committed:
Add acknowledgements in DeveloperGuide

RCPilot1604 committed:
Merge branch 'master' into branch-empty-list-message

# Conflicts:
#	src/main/java/tutorlink/command/ListStudentCommand.java

RCPilot1604 committed:
Merge branch 'master' into branch-max-score-validation

# Conflicts:
#	src/main/java/tutorlink/command/AddComponentCommand.java

RCPilot1604 committed:
Merge pull request #208 from RCPilot1604/branch-fix-193

Add handling of multiple prefixes

RCPilot1604 committed:
Merge pull request #205 from RCPilot1604/branch-fix-173

Make error message when inputted score > max_score more informative

RCPilot1604 committed:
Revert "Merge pull request #205 from RCPilot1604/branch-fix-173"

This reverts commit c48ec21c7265ae73b0de6428f94305852b00f544, reversing
changes made to 1ba047193eacc7e87bbf45196408561651c28c17.

RCPilot1604 committed:
Reapply "Merge pull request #205 from RCPilot1604/branch-fix-173"

This reverts commit 0e22df63cd3eab8ee5282e83b48fddc66f8e424e.

yeekian committed:
Merge pull request #206 from jinzihan2002/Zihan-IOTest

Update text-ui-test to test loading of erroneous data

RCPilot1604 committed:
Merge branch 'master' into branch-empty-list-message

# Conflicts:
#	src/main/java/tutorlink/command/ListStudentCommand.java

RCPilot1604 committed:
Merge pull request #204 from RCPilot1604/branch-empty-list-message

Make empty Student and Component List messages more informative

yeekian committed:
Indication deletion of objects after use in diagrams

RCPilot1604 committed:
Modify UG to lay out further conditions on input commands

RCPilot1604 committed:
Modify component query list to match exactly

This is to prevent cases where grades cannot be added due to components whose
descriptions are substrings of each other.

yeekian committed:
Update png diagrams

yeekian committed:
Correct spelling in DeveloperGuide

yeekian committed:
Add table of contents in DeveloperGuide

yeekian committed:
Update acknowledgements in DeveloperGuide

yeekian committed:
Remove excess lifeline of DeleteStudentCommand diagram

jinzihan2002 committed:
Merge pull request #212 from yeekian/branch-PEDryRunBugFix

Fix developer guide issues

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

yeekian committed:
Merge branch 'master' of https://github.com/yeekian/tp into branch-UserGuide

yeekian committed:
Refactor code to pass CI

RCPilot1604 committed:
Modify ComponentListTest

RCPilot1604 committed:
Fix merge commit

jinzihan2002 committed:
Merge pull request #214 from RCPilot1604/branch-find-student

UG (find_student) and logic changes (ComponentList.findComponent)

yeekian committed:
Merge pull request #215 from yeekian/branch-UserGuide

Refactor Ui class to pass CI

yeekian committed:
Remove redundant setting of matric number to upper case

TrungBui32 committed:
Add HELP_MESSAGE

TrungBui32 committed:
Add helpcommand into Parser

TrungBui32 committed:
Create HelpCommand

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp into branch-UserGuide

TrungBui32 committed:
Fix checkStyle

TrungBui32 committed:
Fix checkStyle second time

TrungBui32 committed:
Add newline at the end

RCPilot1604 committed:
Modify UG to specify rounding and example for decimal

TrungBui32 committed:
Remove redundant break lines

yeekian committed:
Update User Guide

yeekian committed:
Ensure upper case matric number input

yeekian committed:
Update text-ui-test expected outcome

jinzihan2002 committed:
Merge pull request #216 from TrungBui32/helpCommand

Help command

yeekian committed:
Add help information in UserGuide

yeekian committed:
Add table of contents in UserGuide

RCPilot1604 committed:
Add more descriptive NULL_PARAMETER_ERROR messages

RCPilot1604 committed:
Update tests

jinzihan2002 committed:
Merge pull request #213 from RCPilot1604/branch-max-score-validation

Branch max score validation

jinzihan2002 committed:
Merge pull request #217 from RCPilot1604/branch-fix-178

Modify UG to specify rounding and example for decimal

RCPilot1604 committed:
Update AddStudentCommandTest for both null

RCPilot1604 committed:
Fix codestyle violatons

yeekian committed:
Update architecture diagram

yeekian committed:
Add user input constraint description in UserGuide

yeekian committed:
Merge pull request #218 from yeekian/branch-UserGuide

Modify user guide according to issues from dry pe run

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

jinzihan2002 committed:
Update GradeStorage uml

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-DG-2

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Add user input constraint description in UserGuide
  Update architecture diagram
  Add table of contents in UserGuide
  Add help information in UserGuide
  Update text-ui-test expected outcome
  Ensure upper case matric number input
  Update User Guide
  Remove redundant setting of matric number to upper case

jinzihan2002 committed:
Edit DG to fix issues

jinzihan2002 committed:
Merge pull request #219 from RCPilot1604/branch-fix-165

Add more specific null parameter passed exception messages

jinzihan2002 committed:
Add check to prevent excessively large maxScore from being loaded

jinzihan2002 committed:
Add tests for data file error handling in text-ui-test

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-DG-2

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Fix codestyle violatons
  Update AddStudentCommandTest for both null
  Update tests
  Add more descriptive NULL_PARAMETER_ERROR messages

jinzihan2002 committed:
Merge pull request #220 from jinzihan2002/Zihan-DG-2

Edit DG for Storage to fix issues

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-Test-Storage

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Edit DG to fix issues
  Update GradeStorage uml
  Fix codestyle violatons
  Update AddStudentCommandTest for both null
  Update tests
  Add more descriptive NULL_PARAMETER_ERROR messages

jinzihan2002 committed:
Fix checkstyle

ThienDuc3112 committed:
Update thienduc3112 PPP

ThienDuc3112 committed:
Merge pull request #222 from ThienDuc3112/updatePPP

ThienDuc3112 committed:
Edit aboutus.md for github page

ThienDuc3112 committed:
Merge pull request #223 from ThienDuc3112/editPage

jinzihan2002 committed:
Update PPP

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-PPP-2

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Edit aboutus.md for github page
  Update thienduc3112 PPP

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-Test-Storage

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Edit aboutus.md for github page
  Update thienduc3112 PPP

jinzihan2002 committed:
Merge pull request #221 from jinzihan2002/Zihan-Test-Storage

Update text-ui-test to test file error handling of Storage

jinzihan2002 committed:
Merge branch 'master' of github.com:AY2425S1-CS2113-W13-4/tp into Zihan-PPP-2

* 'master' of github.com:AY2425S1-CS2113-W13-4/tp:
  Fix checkstyle
  Add tests for data file error handling in text-ui-test
  Add check to prevent excessively large maxScore from being loaded

jinzihan2002 committed:
Update PPP

jinzihan2002 committed:
Merge pull request #224 from jinzihan2002/Zihan-PPP-2

Update Zihan PPP

jinzihan2002 committed:
Update Setup.png with clearer image

jinzihan2002 committed:
Remove outdated page breaks

jinzihan2002 committed:
Merge pull request #225 from jinzihan2002/Zihan-12-Nov

Increase Setup.png clarity and remove outdated page breaks

RCPilot1604 committed:
Update Ethan PPP

RCPilot1604 committed:
Merge pull request #226 from RCPilot1604/branch-PPP-update

Update Ethan PPP

yeekian committed:
Update picture

yeekian committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-4/tp

# Conflicts:
#	docs/AboutUs.md

yeekian committed:
Update AboutUs

yeekian committed:
Update working links for AboutUs

yeekian committed:
Merge pull request #227 from yeekian/master

Update AboutUs

RCPilot1604 committed:
Update UG

RCPilot1604 committed:
Update Ethan-PPP

RCPilot1604 committed:
Merge branch 'master' into branch-UG-update

RCPilot1604 committed:
Update README.md

RCPilot1604 committed:
Update README.md

RCPilot1604 committed:
Refine add_grade UG

RCPilot1604 committed:
Merge pull request #228 from RCPilot1604/branch-UG-update

Branch UG update

RCPilot1604 committed:
Update UserGuide.md

RCPilot1604 committed:
Merge pull request #229 from AY2425S1-CS2113-W13-4/RCPilot1604-patch-1

Update UserGuide.md

RCPilot1604 committed:
Update UserGuide.md

RCPilot1604 committed:
Merge pull request #230 from AY2425S1-CS2113-W13-4/RCPilot1604-patch-2

Update UserGuide.md

RCPilot1604 committed:
Fix update_component

RCPilot1604 committed:
Merge branch 'master' into fix-update-component-error

ThienDuc3112 committed:
Merge pull request #231 from RCPilot1604/fix-update-component-error

RCPilot1604 committed:
Update AboutUs.md

RCPilot1604 committed:
Merge pull request #1 from RCPilot1604/RCPilot1604-patch-1

Update AboutUs.md

RCPilot1604 committed:
Merge pull request #232 from RCPilot1604/fix-PPP-link

Fix ppp link

RCPilot1604 committed:
Add Javadoc to AddStudentCommand

RCPilot1604 committed:
Add Javadoc to DeleteComponentCommand

RCPilot1604 committed:
Add javadoc to DeleteStudentCommand

RCPilot1604 committed:
Add javadoc to FindStudentCommand

RCPilot1604 committed:
Merge pull request #233 from RCPilot1604/branch-more-javadoc

Branch more javadoc

yeekian committed:
Update PPP

yeekian committed:
Refactor PPP

yeekian committed:
Merge pull request #234 from yeekian/master

Update yeekian PPP

TrungBui32 committed:
Update PPP trungbui32.md

RCPilot1604 committed:
Bugfix for incorrect error message

TrungBui32 committed:
Update PPP

TrungBui32 committed:
Merge pull request #236 from TrungBui32/updatedppp

Updatedppp

RCPilot1604 committed:
Merge pull request #235 from RCPilot1604/branch-emergency-bugfix

Bugfix for incorrect error message

jinzihan2002 committed:
Add author tags

jinzihan2002 committed:
Merge pull request #237 from jinzihan2002/Zihan-authortag

Add author tags for GradeStorage

RCPilot1604 committed:
Modify puml diagram

RCPilot1604 committed:
Fix puml diagram (CRLF to LF)

RCPilot1604 committed:
Merge pull request #239 from RCPilot1604/branch-fix-reposense

Branch fix reposense
